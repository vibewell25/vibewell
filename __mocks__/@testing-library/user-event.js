const userEventMock = () => {
  return {
    // Basic interactions
    click: jest.fn().mockImplementation(() => Promise.resolve()),
    dblClick: jest.fn().mockImplementation(() => Promise.resolve()),
    tripleClick: jest.fn().mockImplementation(() => Promise.resolve()),
    hover: jest.fn().mockImplementation(() => Promise.resolve()),
    unhover: jest.fn().mockImplementation(() => Promise.resolve()),
    
    // Keyboard interactions
    tab: jest.fn().mockImplementation(({ shift } = {}) => Promise.resolve()),
    keyboard: jest.fn().mockImplementation((text) => Promise.resolve()),
    type: jest.fn().mockImplementation((element, text, options) => Promise.resolve()),
    clear: jest.fn().mockImplementation((element) => Promise.resolve()),
    
    // Form interactions
    selectOptions: jest.fn().mockImplementation((element, values) => Promise.resolve()),
    deselectOptions: jest.fn().mockImplementation((element, values) => Promise.resolve()),
    upload: jest.fn().mockImplementation((element, files) => Promise.resolve()),
    
    // Clipboard interactions
    copy: jest.fn().mockImplementation(() => Promise.resolve()),
    cut: jest.fn().mockImplementation(() => Promise.resolve()),
    paste: jest.fn().mockImplementation((text) => Promise.resolve()),
    
    // Pointer interactions
    pointer: jest.fn().mockImplementation(([{ keys, target }]) => Promise.resolve()),
    
    // Setup methods
    setup: jest.fn().mockImplementation((options) => userEventMock()),
    
    // Special interactions for elements
    selectFile: jest.fn().mockImplementation((element, files) => Promise.resolve()),
    
    // Utility methods for testing
    isPointerMovementEvent: jest.fn().mockReturnValue(false),
    isPointerDownEvent: jest.fn().mockReturnValue(false),
    isPointerUpEvent: jest.fn().mockReturnValue(false)
// Export as both default export and named export to handle different import styles
const setup = (options) => userEventMock();

module.exports = {
  __esModule: true,
  default: userEventMock,
  setup
