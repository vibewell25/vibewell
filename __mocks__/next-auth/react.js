const defaultSession = {
  user: { 
id: 'test-user-id', 
    name: 'Test User', 
    email: 'test@example.com',
image: 'https://example.com/test-user.jpg',
    role: 'user'
expires: new Date(Date.now() + 24 * 60 * 60 * 1000).toISOString()
// Mock hooks and functions
const useSession = jest.fn().mockReturnValue({
  data: defaultSession,
  status: 'authenticated',
  update: jest.fn().mockResolvedValue(defaultSession)
const getSession = jest.fn().mockResolvedValue(defaultSession);

const signIn = jest.fn().mockImplementation((provider, options) => {
  return Promise.resolve({ ok: true, error: null });
const signOut = jest.fn().mockImplementation(() => {
  return Promise.resolve({ ok: true });
const getCsrfToken = jest.fn().mockResolvedValue('mock-csrf-token');
const getProviders = jest.fn().mockResolvedValue({
  google: { id: 'google', name: 'Google', type: 'oauth' },
  github: { id: 'github', name: 'GitHub', type: 'oauth' },
  credentials: { id: 'credentials', name: 'Credentials', type: 'credentials' }
// SessionProvider component mock
const SessionProvider = ({ children }) => children;

module.exports = {
  useSession,
  getSession,
  signIn,
  signOut,
  getCsrfToken,
  getProviders,
  SessionProvider
