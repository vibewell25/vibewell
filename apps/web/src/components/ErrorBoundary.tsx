import React, { Component, ErrorInfo, ReactNode } from 'react';

interface ErrorBoundaryProps {
  children: ReactNode;
  fallback?: ReactNode;
  onError?: (error: Error, errorInfo: ErrorInfo) => void;
  onReset?: () => void;
}

interface ErrorBoundaryState {
  hasError: boolean;
}

/**
 * ErrorBoundary component that catches render and event-handler errors
 */
export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): ErrorBoundaryState {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.error('Error caught by ErrorBoundary:', error, errorInfo);
    if (this.props.onError) {
      this.props.onError(error, errorInfo);
    }
  }

  componentDidMount() {
    window.addEventListener('error', this.handleWindowError);
    window.addEventListener('unhandledrejection', this.handlePromiseRejection);
  }

  componentWillUnmount() {
    window.removeEventListener('error', this.handleWindowError);
    window.removeEventListener('unhandledrejection', this.handlePromiseRejection);
  }

  private handleWindowError = (event: ErrorEvent) => {
    event.preventDefault();
    this.setState({ hasError: true });
    if (this.props.onError && event.error instanceof Error) {
      this.props.onError(event.error, {} as ErrorInfo);
    }
  };

  private handlePromiseRejection = (event: PromiseRejectionEvent) => {
    const error = event.reason instanceof Error ? event.reason : new Error(String(event.reason));
    this.setState({ hasError: true });
    if (this.props.onError) {
      this.props.onError(error, {} as ErrorInfo);
    }
  };

  /**
   * Reset error state
   */
  private handleReset = () => {
    this.setState({ hasError: false });
    if (this.props.onReset) {
      this.props.onReset();
    }
  };

  render() {
    if (this.state.hasError) {
      if (this.props.fallback) {
        return this.props.fallback;
      }
      return (
        <div role="alert">
          <p>Something went wrong.</p>
          <button onClick={this.handleReset}>Try again</button>
        </div>
      );
    }
    return <>{this.props.children}</>;
  }
}

export default ErrorBoundary; 