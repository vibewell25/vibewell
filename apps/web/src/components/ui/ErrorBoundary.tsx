export { ErrorBoundary } from '@/components/ErrorBoundary';
export { withErrorBoundary } from '@/hooks/useErrorBoundary';
