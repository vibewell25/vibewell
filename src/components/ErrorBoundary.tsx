'use client';

import React, { Component, ErrorInfo, ReactNode } from 'react';

interface ErrorBoundaryProps {
  children: ReactNode;
  fallback?: ReactNode;
  onError?: (error: Error, errorInfo: ErrorInfo) => void;
}

interface ErrorBoundaryState {
  hasError: boolean;
  error: Error | null;
}

/**
 * A React Error Boundary component that catches JavaScript errors in its child component tree,
 * logs those errors, and displays a fallback UI instead of the component tree that crashed.
 */
export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = { 
      hasError: false,
      error: null
    };
  }

  static getDerivedStateFromError(error: Error): ErrorBoundaryState {
    // Update state so the next render will show the fallback UI
    return { hasError: true, error };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
    // Log the error to an error reporting service
    console.error('Error caught by ErrorBoundary:', error, errorInfo);
    
    // Call the onError prop if provided
    if (this.props.onError) {
      this.props.onError(error, errorInfo);
    }
  }

  render(): ReactNode {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      if (this.props.fallback) {
        return this.props.fallback;
      }
      
      return (
        <div className="p-4 rounded-md bg-red-50 border border-red-200 text-red-800">
          <h2 className="text-lg font-semibold mb-2">Something went wrong</h2>
          <p className="mb-4">The application encountered an unexpected error.</p>
          <details className="text-sm">
            <summary className="cursor-pointer">Error details</summary>
            <pre className="mt-2 p-2 bg-red-100 rounded overflow-auto text-xs">
              {this.state.error?.toString()}
            </pre>
          </details>
          <button
            className="mt-4 px-4 py-2 bg-red-600 text-white rounded hover:bg-red-700 transition-colors"
            onClick={() => this.setState({ hasError: false, error: null })}
          >
            Try again
          </button>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary; 