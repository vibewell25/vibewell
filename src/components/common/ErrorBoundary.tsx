import { Component, ErrorInfo, ReactNode } from 'react';

interface ErrorBoundaryProps {
  fallback?: ReactNode;
  children: ReactNode;
  onError?: (error: Error, errorInfo: ErrorInfo) => void;
}

interface ErrorBoundaryState {
  hasError: boolean;
  error: Error | null;
}

class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = {
      hasError: false,
      error: null,
    };
  }

  static getDerivedStateFromError(error: Error): ErrorBoundaryState {
    // Update state so the next render will show the fallback UI.
    return { hasError: true, error };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
    // You can also log the error to an error reporting service
    console.error('ErrorBoundary caught an error:', error, errorInfo);

    // Call the onError callback if provided
    if (this.props.onError) {
      this.props.onError(error, errorInfo);
    }
  }

  render(): ReactNode {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return (
        this.props.fallback || (
          <div className="rounded-md border border-red-100 bg-red-50 p-4">
            <h2 className="text-lg font-medium text-red-800">Something went wrong</h2>
            <p className="mt-2 text-sm text-red-700">
              {this.state.error?.message || 'An unexpected error occurred'}
            </p>
            <button
              type="button"
              onClick={() => this.setState({ hasError: false, error: null })}
              className="mt-3 text-sm font-medium text-red-600 hover:text-red-500"
            >
              Try again
            </button>
          </div>
        )
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
