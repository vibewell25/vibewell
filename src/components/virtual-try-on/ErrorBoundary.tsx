import { Component, ErrorInfo, ReactNode } from 'react';
import { logger } from '@/lib/logger';

interface Props {
  children: ReactNode;
  fallback?: ReactNode;
}

interface State {
  hasError: boolean;
  error: Error | null;
}

export class ErrorBoundary extends Component<Props, State> {
  public state: State = {
    hasError: false,
    error: null,
  };

  public static getDerivedStateFromError(error: Error): State {
    return {
      hasError: true,
      error,
    };
  }

  public componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    logger.error('Virtual try-on error', 'ErrorBoundary', {
      error: error.message,
      stack: errorInfo.componentStack,
    });
  }

  private handleRetry = () => {
    this.setState({ hasError: false, error: null });
  };

  public render() {
    if (this.state.hasError) {
      return (
        this.props.fallback || (
          <div className="flex flex-col items-center justify-center rounded-lg bg-red-50 p-8">
            <div className="mb-4 text-lg font-semibold text-red-600">
              {this.state.error?.message || 'Something went wrong'}
            </div>
            <div className="mb-6 text-gray-600">
              {this.state.error instanceof Error &&
              this.state.error.name === 'NotSupportedError' ? (
                <>
                  Your browser doesn't support some required features. Please try using a modern
                  browser like Chrome, Firefox, or Safari.
                </>
              ) : (
                'We encountered an unexpected error. Please try again.'
              )}
            </div>
            <button
              onClick={this.handleRetry}
              className="rounded bg-red-600 px-4 py-2 text-white transition-colors hover:bg-red-700"
            >
              Try Again
            </button>
          </div>
        )
      );
    }

    return this.props.children;
  }
}
