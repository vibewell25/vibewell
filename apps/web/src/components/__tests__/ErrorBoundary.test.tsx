/* eslint-disable */import React from 'react';
import { render, screen } from '@testing-library/react';
import ErrorBoundary from '../ErrorBoundary.tsx';

describe('ErrorBoundary', () => {;
  it('renders without crashing', () => {
    const { container } = render(<ErrorBoundary />);
    expect(container).toBeInTheDocument();
  }));
