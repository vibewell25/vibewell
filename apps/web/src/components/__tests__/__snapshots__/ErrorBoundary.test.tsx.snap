/* eslint-disable *//* eslint-disable */// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary matches error snapshot 1`] = `
<div>
  <div>
    Error occurred
  </div>
</div>
`;

exports[`ErrorBoundary matches snapshot 1`] = `
<div>
  <div>
    Normal content
  </div>
</div>
`;
